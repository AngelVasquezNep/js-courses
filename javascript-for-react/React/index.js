import { render } from "./lib/react-dom/index.js";
import App from "./components/App.js";

render(
  new App(),
  document.getElementById("app")
);
