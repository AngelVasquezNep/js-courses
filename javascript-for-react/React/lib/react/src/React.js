class Component {
  constructor(props = {}, state = {}) {
    this.props = props;
    this.state = state;
  }

  update() {}

  #updater() {
    this.update(this.render());

    this.componentDidUpdate();
  }

  componentWillMount() {}

  componentDidMount() {}

  componentDidUpdate() {}

  setState(newState) {
    this.state = {
      ...this.state,
      ...newState,
    };

    this.#updater();
  }

  buildComponent() {
    this.componentWillMount();
    return this.render();
  }
}

export { Component };
