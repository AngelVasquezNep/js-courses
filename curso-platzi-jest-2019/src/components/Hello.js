import React from "react";

const Hellow = ({ title }) => <h1>{title}</h1>;

export default Hellow;
