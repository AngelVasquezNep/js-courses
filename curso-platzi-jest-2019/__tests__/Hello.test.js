import React from "react";

import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Hello from "../src/components/Hello";

Enzyme.configure({ adapter: new Adapter() });

const data = { title: "Hi" };

describe("Hello Component", () => {
  test("Hello mounts", () => {
    const wrapper = shallow(<Hello title={data.title} />);

    const h1 = wrapper.find("h1");

    expect(h1.text()).toBe(data.title);
  });

  test("Hello snapshot", () => {
    const wrapper = shallow(<Hello title={data.title} />);

    const h1 = wrapper.find("h1");

    expect(h1.html()).toMatchSnapshot();
  });
});
