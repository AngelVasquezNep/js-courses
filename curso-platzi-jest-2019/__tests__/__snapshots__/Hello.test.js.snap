// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component Hello snapshot 1`] = `"<h1>Hi</h1>"`;
